packages/dependency-extraction-webpack-plugin/index.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * WordPress dependencies
+ */
+const { camelCaseDash } = require( '@wordpress/scripts/utils' );
+
+/**
+ * External dependencies
+ */
+const { createHash } = require( 'crypto' );
+const { ExternalsPlugin } = require( 'webpack' );
+const { RawSource } = require( 'webpack-sources' );
+
+const WORDPRESS_NAMESPACE = '@wordpress/';
+
+class DependencyExtractionWebpackPlugin {
+	constructor( options ) {
+		this.options = Object.assign(
+			{
+				injectPolyfill: false,
+				useDefaults: true,
+			},
+			options
+		);
+		this.externalizedDeps = new Set();
+		this.externalsPlugin = new ExternalsPlugin( 'global', [ this.externalizeWpDeps.bind( this ) ] );
+	}
+
+	externalizeWpDeps( context, request, callback ) {
+		let externRootRequest;
+
+		// Handle via options.requestToExternal first
+		if ( 'function' === typeof this.options.requestToExternal ) {
+			externRootRequest = this.options.requestToExternal( request );
+		}
+
+		// Cascade to default if unhandled and enabled
+		if ( 'undefined' === typeof externRootRequest && this.options.useDefaults !== false ) {
```

Comment:
yoda conditional, isn't this against js coding standards?  Should this be:

```suggestion
		if ( typeof externRootRequest === 'undefined' && this.options.useDefaults !== false ) {
```

Reply:
Happy to make the change, I'm indifferent, but I don't see any mention of yoda conditionals in the handbook: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/

Reply:
huh, I'm sure I saw a `yoda: 'off'` rule in the gb eslint rules at some point in its history but you're right, its not defined in the handbook nor is it declared (either way) in the current eslint rules.  So I guess it doesn't matter.

Reply:
Although there is this section on the preferred way for type checks that applies here: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#type-checks

Reply:
Related: The standards revisions adopted late last year removed Yoda as being prescribed one way or another:

>Proposed Change: This guideline will be removed from the coding standards. While it is still an option for developers to use this pattern, it will not be enforced one way or the other.

https://make.wordpress.org/core/2018/12/18/proposed-revisions-to-javascript-coding-standards/